### ISSUE TITLE
access level naming incompatible with RFC 2578

### ISSUE THREAD

**Issue by grondman** [lextudio/pysnmp#60]
### Expected behavior

After converting an RFC 2578 compliant MIB file to a .py file using PySMI, I expected to be able to "use" the MIB, e.g. by reading a value.

### Actual behavior

A `NoAccessError` is thrown. Further inspection shows that the implementation of `readTest` in `SNMPv2-SMI.py` checks
```python
if self.maxAccess not in ('readonly', 'readwrite', 'readcreate') or ...
```
while the access levels in the RFC are actually called `read-only`, `read-write` and `read-create`, respectively, i.e. _with_ a dash and PySMI simply copies in these values in the `setMaxAccess()` calls, i.e. the check will never pass...

### Detailed steps

Unfortunately, I cannot share the specific MIB, but you should be able to recreate this problem with any MIB file by:
1. Specifying an access level _with_ a dash in the original MIB file.
2. Convert the MIB file to a Python module using PySMI.
3. Use the resulting module with PySNMP.

### Python package information

6.0.11

### Operating system information

Ubuntu 22.04.4 LTS

### Python information

3.12.1

**Comment by grondman** [comment]
It might well be that I simply misunderstood the documentation and that other steps need to be taken to make things work, but this seems like a fundamental difference in the way access levels are named. Then again, the library has been around for years and I can't imagine that no one ever bumped into this behaviour before, so I must be doing something wrong?

**Comment by lextm** [comment]
https://docs.lextudio.com/pysnmp/upgrade.html

You cannot use newer releases of PySMI if you are still with PySNMP 6.0.

**Comment by grondman** [comment]
> You cannot use newer releases of PySMI if you are still with PySNMP 6.0.

@lextm I did not install (a newer release of) PySMI myself. It simply came as a dependency with PySNMP, so perhaps the dependency management of PySNMP should then be improved, as apparently an incompatible version of PySMI was allowed to co-exist with PySNMP 6.0. Even now with the latest on `main`, the dependency is listed as
```
pysmi-lextudio = "^1.3.0"
```
which means that it can still happen in the future that a newer, possibly incompatible, version of PySMI is allowed to be installed with PySNMP.

**Comment by lextm** [comment]
We fixed that dependency check in 6.0.12, but won't be able to do so for all previous releases.
